docbrown/core/src/graph.rs

```diff
@@ -589,21 +564,39 @@ impl<'a> VertexView<'a, TemporalGraph> {
     }
 
     // FIXME: all the functions using global ID need to be changed to use the physical ID instead
-    pub fn outbound(&'a self) -> Box<dyn Iterator<Item = EdgeView<'a, TemporalGraph>> + 'a> {
+    pub fn outbound(&'a self) -> Box<dyn Iterator<Item=EdgeView<'a, TemporalGraph>> + 'a> {
         if let Some(r) = &self.w {
             self.g.outbound_window(self.g_id, r.clone())
         } else {
             self.g.outbound(self.g_id)
         }
     }
 
-    pub fn inbound(&'a self) -> Box<dyn Iterator<Item = EdgeView<'a, TemporalGraph>> + 'a> {
+    pub fn inbound(&'a self) -> Box<dyn Iterator<Item=EdgeView<'a, TemporalGraph>> + 'a> {
         if let Some(r) = &self.w {
             self.g.inbound_window(self.g_id, r.clone())
         } else {
             self.g.inbound(self.g_id)
         }
     }
+
```

Comment:
I don't like the `unwrap` can you make a function `props_safe` where we return an Option<Box

then instead of unwrap just use `?`

Reply:
then call that function and default to an empty iterator